### ISSUE TITLE
Fix/Refactor/Replace the `opening_hours` library

### ISSUE THREAD

**Issue by hb0nd** [organicmaps/organicmaps#7974]
The current opening_hours implementation can't handle many cases present in OM data. This issue has been opened to centralise discussion on replacement or refactoring of the opening_hours library.

The current implementation is (afaik) a Maps.Me conversion of [opening_hours.js](https://github.com/opening-hours/opening_hours.js) to C++.

A replacement library would need to:
1. Be written in C++ for multi-platform support?
2. (please comment/edit with more)

related issues:
- https://github.com/organicmaps/organicmaps/issues/3883
- https://github.com/organicmaps/organicmaps/issues/5805
- https://github.com/organicmaps/organicmaps/issues/4944
- https://github.com/organicmaps/organicmaps/issues/4186
- https://github.com/organicmaps/organicmaps/issues/3457
- https://github.com/organicmaps/organicmaps/issues/3117
- https://github.com/organicmaps/organicmaps/issues/2379
- https://github.com/organicmaps/organicmaps/issues/1612
- https://github.com/organicmaps/organicmaps/issues/219
- https://github.com/organicmaps/organicmaps/issues/8365
- https://github.com/organicmaps/organicmaps/issues/3731

## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/organicmaps) so you can upvote and help fund this issue.
- We receive the funding once the issue is completed & confirmed by you.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/organicmaps/organicmaps/issues/7974">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/organicmaps/organicmaps/issues/7974/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/organicmaps/organicmaps/issues/7974/pledge.svg">
</picture>
</a>

**Comment by RedAuburn** [comment]
maybe we could generate & store the timetable at map generation, instead of at runtime?

**Comment by biodranik** [comment]
> The current implementation is (afaik) a Maps.Me conversion of [opening_hours.js](https://github.com/opening-hours/opening_hours.js) to C++.

Not exactly. Btw, I can't build the referenced js library. Are there any js => C++ converters?

**Comment by map-per** [comment]
> maybe we could generate & store the opening_hours timetable at map generation, instead of at runtime?

I think that approach would cause problems for complicated `opening_hours` values like e.g. `Mo-Fr 09:00-18:00; 2023 Jun 10-2023 Aug 15 Mo-Fr 09:00-20:00; 2023 Jun 10-2023 Aug 15 Sa 08:00-12:00; PH off` (example from the wiki page)

**Comment by RedAuburn** [comment]
> I think that approach would cause problems for complicated `opening_hours` values like e.g. `Mo-Fr 09:00-18:00; 2023 Jun 10-2023 Aug 15 Mo-Fr 09:00-20:00; 2023 Jun 10-2023 Aug 15 Sa 08:00-12:00; PH off` (example from the wiki page)

mm, you might be right.. I guess it's worth checking how performant it is to get the current opening_hours status to see if any refactoring is even worth it

**Comment by RedAuburn** [comment]
> Not exactly. Btw, I can't build the referenced js library. 

strange, the CI in their git is passing.

> Are there any js => C++ converters?

i'm honestly not sure, but it looks like it's possible? https://surma.dev/things/compile-js/

**Comment by matheusgomesms** [comment]
Does this help? https://github.com/westnordost/osm-opening-hours?tab=readme-ov-file#specification

**Comment by o-andras** [comment]
As far as I could confirm in the code, the 3rd party parser implementation already supports several cases that are not supported/shown correctly in the UI (as in #1612 and #8365, for example; I don't know why they're closed because I can still see these in the latest OM release: https://www.openstreetmap.org/node/2195511282).

I was trying to find where/how these are parsed for POIs but unsuccessfully.

**Comment by biodranik** [comment]
Fixed in #13107
